nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -0,0 +1,297 @@
+            .build();
+
+    private static final String DEFAULT_USER_AGENT = "NiFi";
+    private static final Protocol DEFAULT_PROTOCOL = Protocol.HTTPS;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Collections.unmodifiableList(Arrays.asList(
+                SECRET_NAME_PATTERN,
+                REGION,
+                AWS_CREDENTIALS_PROVIDER_SERVICE,
+                TIMEOUT,
+                SSL_CONTEXT_SERVICE
+        ));
+    }
+
+    @Override
+    public List<ParameterGroup> fetchParameters(final ConfigurationContext context) {
+        AWSSecretsManager secretsManager = this.configureClient(context);
+
+        final List<ParameterGroup> groups = new ArrayList<>();
+        final ListSecretsRequest listSecretsRequest = new ListSecretsRequest();
+        final ListSecretsResult listSecretsResult = secretsManager.listSecrets(listSecretsRequest);
+        for (final SecretListEntry entry : listSecretsResult.getSecretList()) {
+            groups.addAll(fetchSecret(secretsManager, context, entry.getName()));
+        }
+
+        return groups;
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(final ConfigurationContext context, final ComponentLog verificationLogger) {
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+
+        try {
+            final List<ParameterGroup> parameterGroups = fetchParameters(context);
+            int parameterCount = 0;
+            for (final ParameterGroup group : parameterGroups) {
+                parameterCount += group.getParameters().size();
+            }
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.SUCCESSFUL)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation(String.format("Fetched %s secret keys as parameters, across %s groups",
+                            parameterCount, parameterGroups.size()))
+                    .build());
+        } catch (final Exception e) {
+            verificationLogger.error("Failed to fetch parameters", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.FAILED)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation("Failed to fetch parameters: " + e.getMessage())
+                    .build());
+        }
+        return results;
+    }
+
+    private List<ParameterGroup> fetchSecret(final AWSSecretsManager secretsManager, final ConfigurationContext context, final String secretName) {
+        final List<ParameterGroup> groups = new ArrayList<>();
+        final Pattern secretNamePattern = context.getProperty(SECRET_NAME_PATTERN).isSet()
+                ? Pattern.compile(context.getProperty(SECRET_NAME_PATTERN).getValue()) : null;
```

Comment:
It looks like this could set `secretNamePattern` to `null`, but `secretNamePattern.matcher()` expects it to be defined. In this case, the default value should be a static default `Pattern` instance.
```suggestion
        final Pattern secretNamePattern = context.getProperty(SECRET_NAME_PATTERN).isSet()
                ? Pattern.compile(context.getProperty(SECRET_NAME_PATTERN).getValue()) : DEFAULT_PATTERN;
```

Reply:
Actually, since this is a required property, I don't think I need to handle the case where it isn't set.  Will remove.